test/ResponseTest.php

```diff
@@ -66,6 +66,63 @@ public function testReasonPhraseDefaultsToStandards()
         $this->assertEquals('Unprocessable Entity', $response->getReasonPhrase());
     }
 
+    public function ianaCodesReasonPhrasesProvider()
+    {
+        $ianaHttpStatusCodes = new \DOMDocument();
+
+        libxml_set_streams_context(
+            stream_context_create(
+                [
+                    'http' => [
+                        'method'  => 'GET',
+                        'timeout' => 30,
+                    ],
+                ]
+            )
+        );
+
+        $ianaHttpStatusCodes->load('https://www.iana.org/assignments/http-status-codes/http-status-codes.xml');
+
+        if (! $ianaHttpStatusCodes->relaxNGValidate(__DIR__ . '/TestAsset/http-status-codes.rng')) {
+            self::fail();
```

Comment:
This should have a useful message, to make it clear why the failure occurred (timeout reaching the validation file and/or invalid content in the file fetched). I can add that at merge, however.